### PR TITLE
[Enhancement] Add system table be_cloud_native_compactions

### PR DESCRIPTION
**Pull request by sduzh**
```
StarRocks> select * from information_schema.be_cloud_native_compactions;
+-------+--------+-----------+---------+---------+------+---------------------+---------------------+----------+--------+
| BE_ID | TXN_ID | TABLET_ID | VERSION | SKIPPED | RUNS | START_TIME          | FINISH_TIME         | PROGRESS | STATUS |
+-------+--------+-----------+---------+---------+------+---------------------+---------------------+----------+--------+
| 12021 |  18045 |     25021 |      97 |       0 |    1 | 2023-04-13 17:27:15 | 2023-04-13 17:27:15 |      100 | OK     |
| 12021 |  18045 |     25022 |      97 |       0 |    2 | 2023-04-13 17:27:15 | NULL                |       25 | OK     |
| 12021 |  18045 |     25023 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25024 |      97 |       0 |    1 | 2023-04-13 17:27:15 | NULL                |        0 | OK     |
| 12021 |  18045 |     25025 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25026 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25027 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25028 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25029 |      97 |       0 |    1 | 2023-04-13 17:27:15 | NULL                |       33 | OK     |
| 12021 |  18045 |     25030 |      97 |       0 |    1 | 2023-04-13 17:27:15 | NULL                |        0 | OK     |
| 12021 |  18045 |     25031 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25032 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25033 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
| 12021 |  18045 |     25034 |      97 |       0 |    2 | 2023-04-13 17:27:15 | 2023-04-13 17:27:20 |      100 | OK     |
+-------+--------+-----------+---------+---------+------+---------------------+---------------------+----------+--------+
```

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|15|15|100.00%|[]|

File: gensrc/thrift/Descriptors.thrift [hunk 1]
```diff
@@ -148,6 +148,7 @@ enum TSchemaTableType {
     SCH_BE_THREADS,
     SCH_BE_LOGS,
     SCH_BE_BVARS,
+    SCH_BE_CLOUD_NATIVE_COMPACTIONS
```
Comment (kevincai) [inline gensrc/thrift/Descriptors.thrift:151]: have additional `,` at the end?

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java [hunk 2]
```diff
@@ -857,7 +873,9 @@ public enum SchemaTableType {
         SCH_BE_COMPACTIONS("BE_COMPACTIONS", "BE_COMPACTIONS", TSchemaTableType.SCH_BE_COMPACTIONS),
         SCH_BE_THREADS("BE_THREADS", "BE_THREADS", TSchemaTableType.SCH_BE_THREADS),
         SCH_BE_LOGS("BE_LOGS", "BE_LOGS", TSchemaTableType.SCH_BE_LOGS),
-        SCH_BE_BVARS("BE_BVARS", "BE_BVARS", TSchemaTableType.SCH_BE_BVARS);
+        SCH_BE_BVARS("BE_BVARS", "BE_BVARS", TSchemaTableType.SCH_BE_BVARS),
+        SCH_BE_CLOUD_NATIVE_COMPACTIONS("BE_CLOUD_NATIVE_COMPACTIONS", "BE_CLOUD_NATIVE_COMPACTIONS",
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:877]: if the plan is, be will be split into cn + dn, is this `BE_` naming still ok?

**Review by kevincai — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java (hunk 2 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:877]: > if the plan is, be will be split into cn + dn, is this `BE_` naming still ok?

I'm not sure, the naming convention was restricted by the current system table design.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:877]: > > if the plan is, be will be split into cn + dn, is this `BE_` naming still ok?
> 
> I'm not sure, the naming convention was restricted by the current system table design.

Some codes rely on the table name to do some special logics.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/SchemaTable.java:877]: > if the plan is, be will be split into cn + dn, is this `BE_` naming still ok?

@imay How do you think about this?

File: gensrc/thrift/Descriptors.thrift (hunk 1 above)
Comment (sduzh) [reply gensrc/thrift/Descriptors.thrift]: updated

**Review by kevincai — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 15 / 15 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/SchemaTable.java|15|15|100.00%|[]|

File: be/src/exec/schema_scanner/schema_be_cloud_native_compactions_scanner.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        {"BE_ID", TYPE_BIGINT, sizeof(int64_t), false},
+        {"TXN_ID", TYPE_BIGINT, sizeof(int64_t), false},
+        {"TABLET_ID", TYPE_BIGINT, sizeof(int64_t), false},
+        {"VERSION", TYPE_BIGINT, sizeof(int64_t), false},
+        {"SKIPPED", TYPE_BOOLEAN, sizeof(bool), false},
+        {"RUNS", TYPE_INT, sizeof(int), false},
+        {"START_TIME", TYPE_DATETIME, sizeof(DateTimeValue), true},
+        {"FINISH_TIME", TYPE_DATETIME, sizeof(DateTimeValue), true},
+        {"PROGRESS", TYPE_INT, sizeof(int32_t), false},
+        {"Status", TYPE_VARCHAR, sizeof(StringValue), false}};
```
[the remark this answers is not in the document]
Comment (sduzh) [reply be/src/exec/schema_scanner/schema_be_cloud_native_compactions_scanner.cpp]: Fixed

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**